tools/FhirSchemaManager/Exceptions/SchemaManagerException.cs

```diff
@@ -0,0 +1,19 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+
+namespace FhirSchemaManager.Exceptions
+{
+    public class SchemaManagerException : Exception
+    {
+        public SchemaManagerException(string message)
+            : base(message)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(message, nameof(message));
```

Comment:
Perhaps use `Debug.Assert(!string.IsNullOrEmpty(message), "Exception message should not be empty");` we want to avoid the possibility of raising an exception in the constructor of an exception